apps/api/src/friendships/friendships.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { Friendships as friendshipsEntity } from 'types';
+import { User } from "types";
+
+
+@Injectable()
+export class FriendshipsService {
+    constructor (
+        @InjectRepository(User)
+        private readonly userRepository : Repository<User>,
+        @InjectRepository(friendshipsEntity)
+        private readonly friendshipsRepository : Repository<friendshipsEntity>
+    ) {}
+
+    async add (initiator_id : string, target_id : string) : Promise<number> {
+
+
+        const initiatorFriend = await this.friendshipsRepository.findBy({initiator_id : initiator_id});
+        const targetFriend = await this.friendshipsRepository.findBy({target_id : initiator_id});
+
+        const targetAlreadyAsk = targetFriend.filter(e => e.initiator_id === target_id);
+        const userAlreadyAsk = (initiatorFriend).filter(e => e.target_id === target_id);
+
+        if ((targetAlreadyAsk.length !== 0 && targetAlreadyAsk[0].accepted) || (userAlreadyAsk.length !== 0))
+            return 0;
+        else if (targetAlreadyAsk.length !== 0)
+            return this.update(target_id, initiator_id);
+        //else if (userAlreadyAsk.length !== 0)
+        //    return this.update(initiator_id, target_id);
+
+        const newFriendship = new friendshipsEntity();
+        newFriendship.initiator_id = initiator_id;
+        newFriendship.target_id = target_id;
+        newFriendship.accepted = false;
+
+        this.friendshipsRepository.save(newFriendship);
+
+        return 1;
+    }
+
+    async getFriendsList ( user_id : string ) : Promise<{friend : User | null , accept: boolean, ask: boolean}[]> {
+
+        let friendsList :  (User| null)[] = [];
+        let ids : string[] = [];
+        let response : {friend : User | null , accept: boolean, ask: boolean}[] = [];
+
+        const initiatorArray = await this.friendshipsRepository.findBy({initiator_id : user_id});
+        const targetArray = await this.friendshipsRepository.findBy({target_id : user_id});
+        
+        initiatorArray.forEach(e => ids.push(e.target_id));
+        targetArray.forEach(e => ids.push(e.initiator_id));
+
+        const queryBuilder = this.userRepository?.createQueryBuilder().select('*');
```

Comment:
Useless, there are some great TypeOrm methods that do the job. The issue is you did not link your entity with the user Entity but with a foreign key, that's why you are struggling